inference/enterprise/device_manager/metrics_service.py

```diff
@@ -0,0 +1,84 @@
+import time
+import requests
+
+from inference.core.logger import logger
+from inference.core.cache import cache
+from inference.core.version import __version__
+from inference.core.managers.metrics import get_model_metrics, get_system_info
+from inference.core.devices.utils import GLOBAL_DEVICE_ID
+from inference.enterprise.device_manager.container_service import ContainerService
+from inference.core.env import (
+    API_KEY,
+    METRICS_INTERVAL,
+    METRICS_URL,
+    TAGS,
+)
+
+
+class MetricsService:
+    def __init__(self):
+        self.container_service = ContainerService()
+
+    def get_model_ids_by_server_id(self):
+        now = time.time()
+        start = now - METRICS_INTERVAL
+        models = cache.zrangebyscore("models", min=start, max=now)
+        model_ids_by_server_id = dict()
+        for model in models:
+            server_id, model_id = model.split(":")
+            if server_id not in model_ids_by_server_id:
+                model_ids_by_server_id[server_id] = []
+            model_ids_by_server_id[server_id].append(model_id)
+        return model_ids_by_server_id
+
+    def aggregate_model_stats(self, container_id):
+        now = time.time()
+        start = now - METRICS_INTERVAL
+        model_ids = self.get_model_ids_by_server_id().get(container_id, [])
+        models = []
+        for model_id in model_ids:
+            model_metrics = get_model_metrics(
+                container_id, model_id, min=start, max=now
+            )
+            models.append(model_metrics)
+        return models
+
+    def build_container_stats(self):
+        containers = []
+        for id in self.container_service.get_container_ids():
+            container = self.container_service.get_container_by_id(id)
+            if container:
+                container_stats = {}
+                container_stats["uuid"] = container.id
+                # TODO get real container startup_time
+                container_stats["startup_time"] = str(int(time.time()))
```

Comment:
@robiscoding Will `startup_time` get overwritten everytime we aggregate metrics? Should this value come from docker and be the actual time when the container was started?

Reply:
Oh I'll add the real timestamp, this was just a temporary placeholder